### DISCUSSION TITLE
How can I get the Pair Extraordinaire badge?

### DISCUSSION THREAD

**Discussion by iroshancodex** [community/community#109153] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

What are the prerequisites to get it?

**Comment by MoldyLunchBox** [comment, marked as answer]
you can obtain this badge by Coauthoring in a merged pull request. (Requires GitHub Desktop App)

**Comment by kyuna0312** [comment]
Use Github Desktop App and coauth with your github main account and  Desktop app account and same project and make new  pr from your desktop app and merge it and deploy that's enough :3

**Comment by Shahnawazk7944** [comment]
hey @IT21229084 

I believe I am the right person to answer this discussion as I have unlocked all four levels of this badge. To earn the Pair Extraordinaire badge, you must successfully co-author a commit (a commit made by two or more people collaborating) that is merged into the default branch of a repository. How can you achieve this?

1. You need a repository that can be your own repository or one you have write access to. Remember, both you and your collaborator need to have write access to the repository. let suppose your own repository on which you already have write access, now consider your friend also has a GitHub account just give him write access to your repository. How?, Go to  your repository -> Settings -> Collaborators -> Add People -> Enter your friends GitHub username your friends username will show and select it, ask your friend to accept the Collaborators request or copy and send the invitation link to him that's it.

2. Now create a new branch from the default branch (master or main) and make some changes to the code in the new branch. 

3. This is an Important step before committing your changes, add your collaborator(your friend) as a co-author with the commit. There are two ways to do that.

- Using GitHub Desktop :- Download and install GitHub Desktop In the bottom left corner you can see the commit dialog, look for the "Co-author" section and add your collaborator's(your friend) username.

- Using Command Line :- Use the `-c` flag with the `git commit` command. For example, `git commit -m "My changes" -c "collaborators(your friend) username>"`

4. In the final step, commit your changes in your new branch using any of the methods mentioned above, ensuring that the co-author (your friend) information is included. Push the changes to the remote repository, create a pull request, and merge it into the default branch of the repository. That's all you need to do to earn your Pair Extraordinaire badge.


References :- 
https://github.com/orgs/community/discussions/20467
https://docs.github.com/articles/creating-a-commit-with-multiple-authors

**Comment by rirefat** [comment]
To earn the "Pair Extraordinaire" badge on GitHub, you need to participate in a substantial number of pull requests where you've collaborated with other contributors effectively. This badge is awarded to users who consistently contribute high-quality code in pairs or small groups.

Here are some tips to increase your chances of earning the "Pair Extraordinaire" badge:

1. **Collaborate Frequently**: Engage in collaborative projects and actively participate in pull requests where you work closely with other contributors. This demonstrates your ability to collaborate effectively with others.

2. **High-Quality Contributions**: Focus on making meaningful contributions to the projects you're involved in. Write clear, well-documented code, and adhere to the project's coding standards and guidelines.

3. **Effective Communication**: Communicate openly and effectively with your collaborators. Provide constructive feedback, ask questions when needed, and be responsive to comments and suggestions from other contributors.

4. **Consistency**: Make consistent contributions over time. The more frequently you collaborate and contribute to pull requests, the better your chances of earning the badge.

5. **Positive Interactions**: Foster a positive and supportive atmosphere within the project community. Encourage teamwork, recognize the contributions of others, and be respectful in your interactions with fellow contributors.

6. **Review and Improve**: Participate in code reviews and offer valuable feedback to your peers. Actively seek opportunities to learn from others and improve your own skills.

Remember that the "Pair Extraordinaire" badge is awarded based on a combination of factors, including the quality and quantity of your contributions, your collaboration skills, and your overall impact on the project. Keep striving to make valuable contributions and collaborate effectively with others, and you'll be on your way to earning the badge.

**Reply by queenofcorgis** [reply]
@rirefat 
We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by queenofcorgis** [comment]
Hi @IT21229084  

This post has been closed and locked because of version of it has already been posted. We encourage you to use our search bar before posting to find if a similar discussion exists so we don't overwhelm the community with duplicate discussions. Thanks for understanding!

The Community Discussions Team
